File: pset2/exercises/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

void grade (string text);

int main (void)
{
    string text = get_string("Text: ");
    grade(text);
}

void grade (string text)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int grade;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(isalpha(text[i]))
        {
            letters++;
        }
        if(isspace(text[i]))
        {
            words++;
        }
        if(text[i] == 33 || text[i] == 63 || text[i] == 46)
        {
            sentences++;
        }
    }
    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;
    grade = round((0.0588 * L) - (0.296 * S) - 15.8);
    if(grade < 1)
    {
        printf("Before Grade 1\n");
    }else if(grade > 16)
    {
        printf("Grade 16+\n");
    }else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/exercises/readability.py
from cs50 import get_string

def grade(t):
    letters = 0
    words = 1
    sentences = 0
    for i in range(len(t)):
        if t[i].isalpha():
            letters += 1
        elif t[i].isspace():
            words += 1
        elif t[i] == "." or t[i] == "!" or t[i] == "?":
            sentences += 1
    L = (float(letters) / words) * 100
    S = (float(sentences) / words) * 100
    grade = (0.0588 *L) - (0.296 * S) - 15.8
    grade = round(grade)
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

text = get_string("Text: ")
grade(text)